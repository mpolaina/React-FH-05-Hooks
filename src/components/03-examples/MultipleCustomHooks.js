import React from 'react'
import { useFetch } from '../../hooks/useFetch'
import { useCounter } from '../../hooks/useCounter'

export const MultipleCustomHooks = () => {
  
  // custom Hooks
  const { counter, increment } = useCounter(1)
  const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${ counter }`)
  const { author, quote } = !!data && data[0] // si existe data, extrae la posici√≥n [0]
  
  
  
  return (
    <div className="p-5">
      
        <h2>BreakinBad Quotes <small className="text-info">Custom Hooks</small></h2>
        
        { 
          loading
          ? 
            (
              <div className="alert alert-info text-center">Loading...</div>
            )
          :
            (
              <div className="card bg-light w-50 my-4">
                <div className="card-body">
                  <blockquote className="blockquote">
                    <p className="mb-4"> { quote } </p>
                    <footer className="blockquote-footer">{ author }</footer>
                  </blockquote>
                </div>
              </div>
            )
        }
        
        <button 
            className="btn btn-warning"
            onClick={ increment }
        >Quote me!</button>
        
        
        
        
    </div>
  )
}
