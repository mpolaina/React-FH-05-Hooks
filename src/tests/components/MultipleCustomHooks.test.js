import { shallow } from "enzyme"
import { MultipleCustomHooks } from "../../components/03-examples/MultipleCustomHooks"
import { useCounter } from "../../hooks/useCounter"
import { useFetch } from "../../hooks/useFetch"

// llamado a useFetch y useCounter
jest.mock('../../hooks/useFetch')
jest.mock('../../hooks/useCounter')


describe('Pruebas en MultipleCustomHooks', () => {
  
    beforeEach( () => {
      
      useCounter.mockReturnValue({
        counter: 1,
        increment: () => {}
      })
    })
    
  
    test('Debe mostrarse correctamente', () => {
      
        useFetch.mockReturnValue({
          // retorno por defecto del useFetch
          data: null, loading: true, error: null
        })
      
        const wrapper = shallow(<MultipleCustomHooks/>)
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('Debe de mostrar la informaciÃ³n', () => {
      
        useFetch.mockReturnValue({
          data: [{
            author: 'Manuel',
            quote: 'Hola que tal'
          }
          ],
          loading: false, 
          error: null
        })
        const wrapper = shallow(<MultipleCustomHooks/>)
        expect( wrapper.find('.alert').exists() ).toBe( false )
        expect( wrapper.find('.mb-4').text().trim() ).toBe('Hola que tal')
        expect( wrapper.find('footer').text().trim() ).toBe('Manuel')
        
        // console.log(wrapper.html())
        
    })
    
    
})