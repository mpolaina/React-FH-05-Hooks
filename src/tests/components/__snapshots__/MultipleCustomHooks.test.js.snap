// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en MultipleCustomHooks Debe mostrarse correctamente 1`] = `
<div
  className="p-5"
>
  <h2>
    BreakinBad Quotes 
    <small
      className="text-info"
    >
      Custom Hooks
    </small>
  </h2>
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-warning"
    onClick={[Function]}
  >
    Quote me!
  </button>
</div>
`;
